e2e/util/namespace.go

```diff
@@ -49,20 +75,56 @@ func DeleteNamespace(ctx types.Context, cluster types.Cluster, name string) erro
 		},
 	}
 
-	err := cluster.Client.Delete(ctx.Context(), ns)
+	managed, err := isManagedByRamenE2e(ctx, cluster, ns)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Debugf("Namespace %q not found in cluster %q", name, cluster.Name)
+
+			return nil
+		}
+
+		return err
+	}
+
+	if !managed {
+		log.Debugf("Namespace %q in cluster %q is not managed by ramen-e2e, skipping deletion.",
+			name, cluster.Name)
+
+		return nil
+	}
+
+	err = cluster.Client.Delete(ctx.Context(), ns)
 	if err != nil {
 		if !k8serrors.IsNotFound(err) {
 			return err
 		}
 
 		log.Debugf("Namespace %q not found in cluster %q", name, cluster.Name)
-
-		return nil
```

Comment:
Lets keep this here and add a log for "Deleted ..." like we used for other deletions. It can in another commit since it is not part of this change, just a cleanup of existing code.